pkg/api/server/v1alpha2/records.go

```diff
@@ -125,7 +125,7 @@ func (s *Server) GetRecord(ctx context.Context, req *pb.GetRecordRequest) (*pb.R
 func getRecord(txn *gorm.DB, parent, result, name string) (*db.Record, error) {
 	store := &db.Record{}
 	q := txn.
-		Where(&db.Record{Result: db.Result{Parent: parent, Name: result}, Name: name}).
+		Where(&db.Record{Parent: parent, ResultName: result, Name: name}).
```

Comment:
Please add a comment here explaining why this is using the `ResultName` field  (if I'm not mistaken, it's our indexed foreign key to the Results table?)

It took me a moment to figure out that the `db.Result` object created an unnecessary join in the query :sob:

Reply:
Done @adambkaplan. Just for clarifying if anyone else is reading this thread eventually, there's a composite index (parent, result_name, name) in the records table. The name is an unique value as well (that's why the query was working despite of this issue), but it causes a full database scan if used alone. 
Afaik, Gorm doesn't run a join implicitly to load the object that corresponds to a foreign key, but anyhow, in that query populating the Record.Result field has no effect because we want to find the record that matches the provided parent, resultName and name and not load the association.